src/librustc_codegen_llvm/mono_item.rs

```diff
@@ -47,7 +47,9 @@ impl PreDefineMethods<'tcx> for CodegenCx<'ll, 'tcx> {
         visibility: Visibility,
         symbol_name: &str,
     ) {
-        assert!(!instance.substs.needs_infer() && !instance.substs.has_param_types());
+        if !self.tcx.sess.opts.debugging_opts.polymorphize {
+            assert!(!instance.substs.needs_infer() && !instance.substs.has_param_types());
```

Comment:
`needs_infer` should remain unconditional (and you can probably get rid of `has_param_types`).

Maybe we should audit all the uses of `has_param_types`, I'm starting to get suspicious of it (also, the name doesn't make a lot of sense IMO).

Reply:
Fixed in 5b3c02b.